src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+        /// <param name="value">Integer value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, int value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a float value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Float value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, float value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            if (float.IsInfinity(value) || float.IsNaN(value))
+            {
+                await writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo)).ConfigureAwait(false);
+            }
+            else
+            {
+                // float.ToString() supports a max scale of six,
+                // whereas float.MinValue and float.MaxValue have 8 digits scale. Hence we need
+                // to use XmlConvert in all other cases, except infinity
+                await writer.WriteAsync(XmlConvert.ToString(value)).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Write a short value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Short value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, short value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a long value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Long value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, long value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a double value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Double value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, double value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            if (JsonSharedUtils.IsDoubleValueSerializedAsString(value))
+            {
+                await writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo)).ConfigureAwait(false);

```

Comment:
Add Culture Info in `ToString`
`value.ToString(ODataNumberFormatInfo, CultureInfo.InvariantCulture)`

Reply:
Refer https://github.com/OData/odata.net/pull/2002#discussion_r582865181